Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1174,19 +1183,32 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
         }
 
         let localCollection = try self.localCollection(for: namespace, withType: Document.self)
+        let undoCollection = try self.undoCollection(for: namespace)
+
+        let documentBeforeUpdate = try localCollection.find([idField: documentId], options: nil).next()
+        if let documentBeforeUpdate = documentBeforeUpdate {
+            try undoCollection.insertOne(documentBeforeUpdate)
+        }
+
         // Since we are accepting the remote document as the resolution to the conflict, it may
         // contain version information. Clone the document and remove forbidden fields from it before
         // storing it in the collection.
         let docForStorage = DataSynchronizer.sanitizeDocument(remoteDocument)
         try localCollection.findOneAndReplace(filter: ["_id": documentId], replacement: docForStorage, options: FindOneAndReplaceOptions(upsert: true))
         try config.setPendingWritesComplete(atVersion: atVersion)
+        if documentBeforeUpdate != nil {
+            try undoCollection.deleteOne([idField: documentId])
+        }
 
         let event = ChangeEvent<Document>.changeEventForLocalReplace(namespace: namespace,
                                                                      documentId: documentId,
                                                                      document: docForStorage,
                                                                      writePending: false)
-        self.emitEvent(documentId: documentId,
-                       event: event)
+
+        deferredEventEmittingBlock = {
```

Comment:
[q] Same as above, any reason this is deferred?

Reply:
parity with android

Reply:
I feel like this is such a small detail that parity doesn't need to be maintained.

Reply:
I don't think it's a small detail, this is the only way to emit the event after the lock is unlocked, which is what happens in android

Reply:
Ah I see. Makes sense.